cmake/loki_transform_helpers.cmake

```diff
@@ -84,41 +84,6 @@ macro( _loki_transform_parse_options )
         list( APPEND _ARGS --cpp )
     endif()
 
-    if( _PAR_DATA_OFFLOAD )
```

Comment:
With these removed, the argument definition to `loki_transform` (ll. 48-57 in `loki_transform.cmake`) should be updated accordingly:
```cmake
    set( options CPP )
    set( oneValueArgs COMMAND MODE FRONTEND CONFIG BUILDDIR )
    set( multiValueArgs OUTPUT DEPENDS SOURCES HEADERS INCLUDES DEFINITIONS OMNI_INCLUDE XMOD )
```
Please also update the docstring of `loki_transform` accordingly.

Also, since we only have the `CPP` option left now, it might be worthwhile to just remove this macro and instead inline the remaining 3 lines of code into `loki_transform` and `loki_transform_plan` directly